pkg/core/writer/spans.go

```diff
@@ -18,17 +18,18 @@ func (sw *SignalFxWriter) sendSpans(ctx context.Context, spans []*trace.Span) er
 		sw.serviceTracker.AddSpans(sw.ctx, spans)
 	}
 
-	// This sends synchonously
-	err := sw.client.AddSpans(context.Background(), spans)
-	if err != nil {
-		log.WithFields(log.Fields{
-			"error": err,
-		}).Error("Error shipping spans to SignalFx")
-		// If there is an error sending spans then just forget about them.
-		return err
+	if sw.client != nil {
```

Comment:
Remove this conditional since the Splunk writer isn't going to deal with trace spans at this point.

Reply:
That's fine. Note: I have another branch built on this one that exports spans too: https://github.com/atoulme/signalfx-agent/tree/export_traces_too